### PR TITLE
Recovers separate output dirs for different forcefields

### PR DESCRIPTION
**Pull request by j3mdamas**
Also:
- changes in printing functions
- fixing parameterize tutorial
- minor fixes in long tests

This won't pass tests for now, as I haven't changed the data. @raimis, can you pre-approve what's done, and then I change data to match the changes?

### CODE REVIEW

File: htmd/parameterization/cli.py [hunk 1]
```diff
@@ -481,14 +481,14 @@ def _printEnergies(molecule, parameters, filename):
...
-== Diagnostic Energies ==
+== Diagnostic Energies (in kcal/mol) ==
 
-Bond     : {BOND_ENERGY}
-Angle    : {ANGLE_ENERGY}
-Dihedral : {DIHEDRAL_ENERGY}
-Improper : {IMPROPER_ENERGY}
-Electro  : {ELEC_ENERGY}
-VdW      : {VDW_ENERGY}
+Bond     : {BOND_ENERGY:10.3f}
```
Comment (raimis) [inline htmd/parameterization/cli.py]: We need `.6f` as our test tolerances are set to 1e-6

File: htmd/parameterization/cli.py [hunk 2]
```diff
@@ -481,14 +481,14 @@ def _printEnergies(molecule, parameters, filename):
     energies = FFEvaluate(molecule, parameters).calculateEnergies(molecule.coords[:, :, 0])
 
     string = '''
-== Diagnostic Energies ==
+== Diagnostic Energies (in kcal/mol) ==
```
Comment (raimis) [inline htmd/parameterization/cli.py]: Units at the end of the quantity, not in the header

File: htmd/parameterization/cli.py [hunk 3]
```diff
@@ -497,13 +497,29 @@ def _printEnergies(molecule, parameters, filename):
            ELEC_ENERGY=energies['elec'],
            VDW_ENERGY=energies['vdw'])
 
+    for l in string.split('\n'):
```
Comment (raimis) [inline htmd/parameterization/cli.py]: Human-readable variable name: `l` --> `line`

File: htmd/parameterization/cli.py [hunk 4]
```diff
@@ -497,13 +497,29 @@ def _printEnergies(molecule, parameters, filename):
...
+        logger.info(l)
     with open(filename, 'w') as file_:
         file_.write(string)
     logger.info('Write energy file: {}'.format(filename))
 
-    logger.info('Diagnostic energies:')
-    for name in ('bond', 'angle', 'dihedral', 'improper', 'elec', 'vdw'):
-        logger.info('   {:8s} : {:10.3f} kcal/mol'.format(name, energies[name]))
+
+def _printArguments(args, filename=None):
```
Comment (raimis) [inline htmd/parameterization/cli.py]: Can the function definition be moved  before, `_prepare_molecule` in the file.

File: htmd/parameterization/cli.py [hunk 5]
```diff
@@ -497,13 +497,29 @@ def _printEnergies(molecule, parameters, filename):
...
+
+    if filename:
```
Comment (raimis) [inline htmd/parameterization/cli.py]: Looks a bit over-engineered way to write a text file.

File: htmd/parameterization/cli.py [hunk 6]
```diff
@@ -512,8 +528,13 @@ def _output_results(mol, parameters, original_coords, args):
 
     logger.info('=== Results ===')
 
+    dir = os.path.join(args.outdir, 'parameters', args.forcefield)
```
Comment (raimis) [inline htmd/parameterization/cli.py]: `dir` --> `paramDir`. It shadow built-in `dir`

File: htmd/parameterization/cli.py [hunk 7]
```diff
@@ -549,12 +570,7 @@ def main_parameterize(arguments=None):
     if args.optimize_dihedral is not parser.get_default('optimize_dihedral'):
         raise DeprecationWarning('Use `--scan-type` instead.')
 
-    # Print arguments
-    logger.info('=== Arguments ===')
-    for key, value in vars(args).items():
-        if key in ('fake_qm',):  # Hidden
-            continue
-        logger.info('{:>20s}: {:s}'.format(key, str(value)))        
+    _printArguments(args)
```
Comment (raimis) [inline htmd/parameterization/cli.py]: This should go after the parse.

File: htmd/parameterization/cli.py (hunk 2 above)
Comment (j3mdamas) [reply htmd/parameterization/cli.py]: Any scientific paper disagrees with you, but as you wish

File: htmd/parameterization/cli.py (hunk 3 above)
Comment (j3mdamas) [reply htmd/parameterization/cli.py]: Done.

File: htmd/parameterization/cli.py (hunk 4 above)
Comment (j3mdamas) [reply htmd/parameterization/cli.py]: Sure, no problem.

File: htmd/parameterization/cli.py (hunk 2 above)
Comment (raimis) [reply htmd/parameterization/cli.py]: Just keeping consistent with the other log messages

File: htmd/parameterization/cli.py (hunk 6 above)
Comment (j3mdamas) [reply htmd/parameterization/cli.py]: Correct!

File: htmd/parameterization/cli.py (hunk 7 above)
Comment (j3mdamas) [reply htmd/parameterization/cli.py]: Done

File: htmd/parameterization/cli.py (hunk 5 above)
Comment (j3mdamas) [reply htmd/parameterization/cli.py]: I've added some comments to make it more obvious

**Comment by j3mdamas** [comment]
I've just addressed your comments. Let me know if it's OK so I can change the data accordingly.

**Comment by j3mdamas** [comment]
New data added: https://github.com/Acellera/htmd-data/pull/31

Some fixes to the tests (particular notice to the adding of a new assertion to overcome a "bug/feature" of `zip`). Fixed also some bugs of the cli that were overseen in my previous single test.

If it passes travis, I think it's ready to merge, but feel free to review the fixes I introduced.

@raimis

**Comment by raimis** [comment]
@j3mdamas Great work! Thanks for fixing all the tests.

**Pull request merged**